textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -0,0 +1,167 @@
+package edu.uci.ics.textdb.dataflow.keywordmatch;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import edu.uci.ics.textdb.api.common.Attribute;
+import edu.uci.ics.textdb.api.common.IField;
+import edu.uci.ics.textdb.api.common.IPredicate;
+import edu.uci.ics.textdb.api.common.ITuple;
+import edu.uci.ics.textdb.api.common.Schema;
+import edu.uci.ics.textdb.api.dataflow.IOperator;
+import edu.uci.ics.textdb.api.dataflow.ISourceOperator;
+import edu.uci.ics.textdb.common.exception.DataFlowException;
+import edu.uci.ics.textdb.common.field.Span;
+import edu.uci.ics.textdb.common.field.StringField;
+import edu.uci.ics.textdb.common.field.TextField;
+import edu.uci.ics.textdb.common.utils.Utils;
+import edu.uci.ics.textdb.dataflow.common.KeywordPredicate;
+
+/**
+ *  @author prakul
+ *
+ */
+public class KeywordMatcher implements IOperator {
+    private final KeywordPredicate predicate;
+    private ISourceOperator sourceOperator;
+    private ArrayList<Pattern> patternList;
+    private List<Span> spanList;
+    private  String queryValue;
+    private  List<Attribute> attributeList;
+    private  ArrayList<String> queryValueArray;
+
+
+    public KeywordMatcher(IPredicate predicate, ISourceOperator sourceOperator) {
+        this.predicate = (KeywordPredicate)predicate;
+        this.sourceOperator = sourceOperator;
+    }
+
+    @Override
+    public void open() throws DataFlowException {
+        try {
+            Pattern pattern;
+            String regex;
+
+            sourceOperator.open();
+            queryValue = predicate.getQuery();
+            attributeList = predicate.getAttributeList();
+            queryValueArray = predicate.getTokens();
+            patternList = new ArrayList<Pattern>();
+            for(String token : queryValueArray ){
+                regex = "\\b" + token.toLowerCase() + "\\b";
+                pattern = Pattern.compile(regex);
+                patternList.add(pattern);
+            }
+
+            spanList = new ArrayList<>();
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            throw new DataFlowException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public ITuple getNextTuple() throws DataFlowException {
+        String documentValue;
```

Comment:
A general comment on this function: I see you have moved some of the variables declared in the class to each function, which is very good.  We can do even better: move the variables to the right namespaces or submodules of code.  For instance, this "documentValue" variable can be moved to the space it's used, so that the code is easier to read.